python/simulation.py

```diff
@@ -1671,28 +1676,36 @@ def get_source_slice(self, component, vol=None, center=None, size=None):
         self.fields.get_source_slice(v, component ,arr)
         return arr
 
-    def get_array_metadata(self, vol=None, center=None, size=None, collapse=False):
-         if vol is None and center is None and size is None:
-             v = self.fields.total_volume()
-         else:
-             v = self._volume_from_kwargs(vol, center, size)
-         dims = np.zeros(3, dtype=np.uintp)
-         rank, dirs = mp._get_array_slice_dimensions(self.fields, v, dims, collapse)
-
-         nxyz = np.ones(3, dtype=np.intp)
-         for r in range(0,rank):
-             nxyz[dirs[r]]=dims[r]
-         nw=nxyz[0]*nxyz[1]*nxyz[2]
-         xtics=np.zeros(nxyz[0],dtype=np.float64)
-         ytics=np.zeros(nxyz[1],dtype=np.float64)
-         ztics=np.zeros(nxyz[2],dtype=np.float64)
-         weights=np.zeros(nw,dtype=np.float64)
-         self.fields.get_array_metadata(v, xtics, ytics, ztics, weights, collapse)
-         return (xtics,ytics,ztics,np.reshape(weights,dims[np.nonzero(dims)]))
+    def get_source_slice(self, component, vol=None, center=None, size=None):
+        warnings.warn('get_source_slice is deprecated. Please use get_source instead', DeprecationWarning)
+        return self.get_source(component, vol=vol, center=center, size=size)
+
+    def get_array_metadata(self, vol=None, center=None, size=None, dft=None):
+        if dft:
+            vol = dft_cell.where
+        if vol is None and center is None and size is None:
+            v = self.fields.total_volume()
+        else:
+            v = self._volume_from_kwargs(vol, center, size)
+        dims = np.zeros(3, dtype=np.uintp)
+        rank, dirs = mp._get_array_slice_dimensions(self.fields, v, dims, True)
```

Comment:
Should be `bool(dft)` rather than `True`, I think?

Reply:
Fixed.